src/components/InteractionLayer/index.js

```diff
@@ -92,22 +92,22 @@ class InteractionLayer extends React.Component {
     touchY: null,
   };
 
+  componentDidMount() {
+    if (this.props.ruler.timestamp) {
+      this.setRulerPosition(this.props.ruler.timestamp);
+    }
+  }
+
   componentWillReceiveProps(nextProps) {
     const {
-      subDomainsByItemId: prevSubDomainsByItemId,
+      // FIXME: Migrate this to `subDomainsByItemId`.
+      timeSubDomain: prevTimeSubDomain,
       ruler,
       xScalerFactory,
     } = this.props;
-    const { subDomainsByItemId: nextSubDomainsByItemId, width } = nextProps;
-    const { touchX, touchY } = this.state;
-
     // FIXME: Don't assume a single time domain
```

Comment:
It seems this FIXME can be removed accordingly

Reply:
not quite -- it's still pulling it in from the props. Not as badly, but still present. 😞